Resolve a number of TypeErrors

ArchRBX opened the pull request:
PR fixes some improper typing that generates Type Errors in a `strict` context.

pinehappi commented:
Thanks!!

pinehappi reviewed (commented):
Does this change fix errors in your environment? I had 87 type errors, but they disappeared when I reverted this change (using normal Roblox Studio).

ArchRBX commented:
> Does this change fix errors in your environment? I had 87 type errors, but they disappeared when I reverted this change (using normal Roblox Studio). (Theme script, line 806)

Apologies for the delay! It fixes most of the issues, though I've addressed the remaining type errors in https://github.com/pinehappi/DataDelve/pull/11.
